lib/mayaUsd/fileio/jobs/writeJob.cpp

```diff
@@ -1375,6 +1378,44 @@ bool UsdMaya_WriteJob::_CheckNameClashes(const SdfPath& path, const MDagPath& da
     return true;
 }
 
+void UsdMaya_WriteJob::_AddDefaultPrimAccessibility()
+{
+    auto defaultPrim = mJobCtx.mStage->GetDefaultPrim();
+    if (!defaultPrim) {
+        return;
+    }
+
+    auto accessibilityLabel = mJobCtx.mArgs.accessibilityLabel;
+    auto accessibilityDescription = mJobCtx.mArgs.accessibilityDescription;
+    if (accessibilityLabel.empty() && accessibilityDescription.empty()) {
+        return;
+    }
+
+    /* The USD AccessibilityAPI is only available from OpenUSD 25.5 onwards.
+     * However, it is a very simple API and can be written ad-hoc here.
+     * Once Maya-USD switches to only supporting USD 25.5 and above in the future,
+     * we can replace these with the actual API method calls.
+     */
```

Comment:
We always need to support older USD versions that are tied to a specific Maya version. So we will be supporting < 25.5 for a few years. I think this should use an ifdef against USD version and if the API is present, use it. Alternatively instead of testing directly against a USD version you can look at `cmake/modules/FindUSD.cmake` where we query USD files for USD features and set cmake var when present. Then you can use that in your ifdef instead also requires setting a define (from the cmake var).

Reply:
Hi Dhruv,

1. One way is to call python from MEL to get the USD version. See `mayaUsdTranslatorExport_ShouldEnableSplines` for example.
2. Yes we do. Our preflight run includes a Maya 2026 built with USD 24.11 and another one with 25.5.
3. Yes keep the ad-hoc setup for below 25.5. For 25.5+ we want the real API and for below your ad-hoc way.

Thanks, Sean

P.S. I asked one of our designers to have a look so he might have some comments for you as well.

Reply:
Addressed your notes, and am working on the stage proxy attribute editor notes next.